sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaSuite.scala

```diff
@@ -198,6 +205,31 @@ abstract class ParquetSchemaTest extends ParquetTest with SharedSparkSession {
 }
 
 class ParquetSchemaInferenceSuite extends ParquetSchemaTest {
+  testSchemaInference[Tuple1[Long]](
```

Comment:
Can this case run passed before Spark 3.2, in my impression, Parquet 1.10.1 used by Spark 3.1 does not support nanos type, does it?

Reply:
This particular case doesn’t pass, neither does similar tests for `TIMESTAMP(MILLIS,true)` etc https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaSuite.scala#L2234 due to `No enum constant org.apache.parquet.schema.OriginalType.TIMESTAMP` when trying to convert the message.

https://github.com/apache/spark/blob/branch-3.1/sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaSuite.scala#L65

Reply:
Then is this really a regression?

Reply:
So I've been looking further into it, it's because the message is different between `1.10.1` and `1.12.3`.

In `1.10.1` the message is

```
message schema {
  required int64 attribute;
}
```

where as `1.12.3` the message is the same as the unit test

```
message schema {
  required int64 attribute (TIMESTAMP(NANOS,true));
}
```

So in Spark 3.1.0 you end up hitting this block with returns a `LongType` https://github.com/apache/spark/blob/branch-3.1/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaConverter.scala#L146

where as since 3.2 you hit https://github.com/apache/spark/blob/branch-3.2/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaConverter.scala#L174 because a case for `TimeUnit.NANOS` is not covered

Reply:
@cloud-fan moving Parquet from `1.10.1` to `1.12.3` introduced this regression where Spark 3.1 returned `LongType` and Spark 3.2 fails on illegal type.